jbehave-support-core/src/main/java/org/jbehavesupport/core/internal/web/WebScreenshotCreator.java

```diff
@@ -61,16 +61,32 @@ public int getHierarchy() {
 
     public void createScreenshot(Type screenShotType) {
         if (desiredMode.getHierarchy() >= screenShotType.getHierarchy()) {
-            try {
-                if (ExpectedConditions.alertIsPresent().apply(driver) != null) {
-                    log.info("Can't take screenshot (alert is present)");
-                } else if (driver instanceof TakesScreenshot) {
-                    takesScreenshot(screenShotType);
+            if (isAlertPresent()) {
+                log.info("Can't take screenshot (alert is present)");
+            } else {
+                try {
+                    if (driver instanceof TakesScreenshot) {
+                        takesScreenshot(screenShotType);
+                    }
+                } catch (IOException e) {
+                    throw new UncheckedIOException(e);
                 }
-            } catch (IOException e) {
-                throw new UncheckedIOException(e);
             }
         }
+
+    }
+
+    private boolean isAlertPresent() {
+        boolean alertPresent = false;
+        try {
+            if (driver.switchTo() != null) {
+                driver.switchTo().alert();
+                alertPresent = true;
+            }
+        } catch (NoAlertPresentException x) {
+            // no op
+        }
+        return alertPresent;
```

Comment:
```        try {
            if (driver.switchTo() != null) {
                driver.switchTo().alert();
                return true;
            }
        } catch (NoAlertPresentException ignored) {
        }
        return false;
```

Reply:
`ignored` as a name of the exception doesn't do anything so I left it as is. Removed the variable.